policy/lib/metadata_helper.rego

```diff
@@ -0,0 +1,13 @@
+package lib
+
+import rego.v1
+
+# The first entry in the chain always points to the active rule, even if it has
+# no declared annotations (in which case the annotations member is not present).
+# Thus, result_helper assumes every rule defines annotations. At the very least
+# custom.short_name must be present.
+_rule_annotations(chain) := chain[0].annotations
+
+release_restrictions_apply(chain) if {
```

Comment:
One suggestion about the name of this. We're replacing things like _schedule_restriction_applies and _expires_label_check_applies. Maybe "release" is not the best choice. Since it's about the pipeline_intention, how about this:

```suggestion
pipeline_intention_match(chain) if {
```

or 
```suggestion
pipeline_intention_matches_metadata(chain) if {
```